contracts/pods/PodFactory.sol

```diff
@@ -212,40 +201,43 @@ contract PodFactory is CoreRef, IPodFactory {
     {
         uint256 podId = memberToken.getNextAvailablePodId();
 
-        address safeAddress = createPod(
+        address safeAddress = _createPod(
             _config.members,
             _config.threshold,
             _config.label,
             _config.ensString,
             _config.imageUrl,
-            _config.admin,
+            address(podAdminGateway),
             podId
         );
 
         address timelock;
-        // TODO: Add option to avoid setting a timelock
-        if (minDelay != 0) {
-            // TODO: Enforce min delay if defined
-            require(minDelay >= 1 days) {
-                timelock = createOptimisticTimelock(
-                safeAddress,
-                _config.minDelay,
-                podExecutor,
-                _config.admin
+        if (_config.minDelay != 0) {
+            require(
+                _config.minDelay >= MIN_TIMELOCK_DELAY,
+                "Min delay too small"
+            );
+            timelock = address(
+                _createTimelock(
+                    safeAddress,
+                    _config.minDelay,
+                    address(podExecutor),
+                    address(podAdminGateway)
+                )
             );
-            // Set mapping from podId to timelock for reference
-            getPodTimelock[podId] = timelock;
-            getPodId[timelock] = podId;
-            latestPodId = podId;
-            }
-            
+        } else {
+            timelock = address(0);
         }
```

Comment:
not necessary else condition, empty variables are default initialized to null value 0x0;

Reply:
Good point, removed. Will add a comment to make it explicit